tests/components/tankerkoenig/test_binary_sensor.py

```diff
@@ -0,0 +1,25 @@
+"""Tests for the Tankerkoening integration."""
+
+from __future__ import annotations
+
+import pytest
+from syrupy import SnapshotAssertion
+
+from homeassistant.const import STATE_ON
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.usefixtures("setup_integration")
+async def test_binary_sensor(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test the tankerkoenig binary sensors."""
+
+    state = hass.states.get("binary_sensor.station_somewhere_street_1_status")
+    assert state
+    assert state.state == STATE_ON
+    assert state.attributes == snapshot
```

Comment:
We have a helper for this merged yesterday https://github.com/home-assistant/core/pull/115880

Reply:
mhhh ... to use this, we can't use the `setup_integration` fixture anymore, since we need to limit the integration setup to only one platform. not sure, if this is worth to rearrange all tests 🤔

Reply:
maybe the new `snapshot_platform` should take a paramater for the wanted platform to be snapshoted :think

Reply:
- #115922

Reply:
@jpbede maybe we can go as is for now and if #115922 get's merged, we can make use of it in a separate PR?

Reply:
Sure :)